src/runners/rfqm_worker_runner.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * Runs the RFQM MetaTransaction Consumer
+ */
+import { createMetricsRouter, MetricsService } from '@0x/api-utils';
+import { SQS } from 'aws-sdk';
+import express from 'express';
+import { Counter } from 'prom-client';
+
+import { ENABLE_PROMETHEUS_METRICS, PROMETHEUS_PORT, RFQM_META_TX_SQS_URL } from '../config';
+import { METRICS_PATH } from '../constants';
+import { getDBConnectionAsync } from '../db_connection';
+import { logger } from '../logger';
+import { RfqmService } from '../services/rfqm_service';
+import { SqsClient } from '../utils/sqs_client';
+import { SqsConsumer } from '../utils/sqs_consumer';
+
+import { buildRfqmServiceAsync } from './http_rfqm_service_runner';
+
+const RFQM_JOB_DEQUEUED = new Counter({
+    name: 'rfqm_job_dequeued',
+    help: 'An Rfqm Job was pulled from the queue',
+});
+
+const RFQM_JOB_SUCCEEDED = new Counter({
+    name: 'rfqm_job_succeeded',
+    help: 'An Rfqm Job succeeded',
+});
+
+process.on('uncaughtException', (err) => {
+    logger.error(err);
+    process.exit(1);
+});
+
+process.on('unhandledRejection', (err) => {
+    if (err) {
+        logger.error(err);
+    }
+});
+
+if (require.main === module) {
+    (async () => {
+        // Start the Metrics Server
+        startMetricsServer();
+
+        // Build dependencies
+        const connection = await getDBConnectionAsync();
+        const rfqmService = await buildRfqmServiceAsync(connection);
+
+        // Run the worker
+        await runRfqmWorkerAsync(rfqmService);
+    })().catch((error) => logger.error(error.stack));
+}
+
+/**
+ * Runs the Rfqm Consumer
+ */
+export async function runRfqmWorkerAsync(rfqmService: RfqmService): Promise<SqsConsumer> {
+    // Build the Sqs consumer
+    const sqsClient = new SqsClient(new SQS({ apiVersion: '2012-11-05' }), RFQM_META_TX_SQS_URL!);
+    const consumer = new SqsConsumer({
+        sqsClient,
+        handleMessage: async (message) => {
+            RFQM_JOB_DEQUEUED.inc();
+            const orderHash = message.Body!;
+            logger.info({ orderHash }, 'about to process job');
+            return rfqmService.processRfqmJobAsync(orderHash);
+        },
+        afterHandle: async (message) => {
+            const orderHash = message.Body!;
+            logger.info({ orderHash }, 'job successfully ran');
+            RFQM_JOB_SUCCEEDED.inc();
+        },
+    });
+
+    // Start the consumer - aka the worker
+    consumer.consumeAsync().catch((e) => logger.error(e));
+    logger.info('Rfqm Consumer running');
```

Comment:
```suggestion
    consumer.consumeAsync()
        .then(() =>   logger.info('Rfqm Consumer running'))
        .catch((e) => logger.error(e));
```

Reply:
ahh, so because `consumeAsync` won't get resolved until we call `consumer.stop` (and for our purposes, we don't want the consumer to ever stop), the `then` block isn't expected to run. (nor is the `catch` block for that matter, but it's there just in case)

The logger line follows the `consumeAsync` invocation - so we start the infinite consume loop, and then we immediately log that we're starting